lib/Runtime/Library/JsBuiltInEngineInterfaceExtensionObject.cpp

```diff
@@ -0,0 +1,410 @@
+        scriptFunction->SetPropertyWithAttributes(PropertyIds::caller, library->nullValue, PropertyNone, nullptr);
+        scriptFunction->SetPropertyWithAttributes(PropertyIds::arguments, library->nullValue, PropertyNone, nullptr);
+
+        scriptFunction->SetConfigurable(PropertyIds::prototype, true);
+        scriptFunction->DeleteProperty(PropertyIds::prototype, Js::PropertyOperationFlags::PropertyOperation_None);
+
+        if (scriptFunction->GetScriptContext()->GetConfig()->IsES6FunctionNameEnabled())
+        {
+            scriptFunction->SetPropertyWithAttributes(PropertyIds::name, methodName, PropertyConfigurable, nullptr);
+        }
+
+        library->AddMember(prototype, functionIdentifier, scriptFunction);
+
+        if (!JavascriptOperators::IsUndefinedOrNull(aliasProperty))
+        {
+            JavascriptString * alias = JavascriptConversion::ToString(aliasProperty, scriptContext);
+            // Cannot do a string to property id search here, Symbol.* have different hashing mechanism, so resort to this str compare
+            PropertyIds aliasFunctionIdentifier = alias->BufferEquals(_u("Symbol.iterator"), 15) ? PropertyIds::_symbolIterator :
+                JavascriptOperators::GetPropertyId(alias, scriptContext);
+            library->AddMember(prototype, aliasFunctionIdentifier, scriptFunction);
+        }
+
+        if (functionIdentifier == PropertyIds::values)
+        {
+            library->arrayPrototypeDefaultValuesFunction = func;
+        }
+
+        //Don't need to return anything
+        return library->GetUndefined();
+    }
+
+    Var JsBuiltInEngineInterfaceExtensionObject::EntryJsBuiltIn_Internal_GetArrayLengthFunction(RecyclableObject *function, CallInfo callInfo, ...)
+    {
+        EngineInterfaceObject_CommonFunctionProlog(function, callInfo);
+
+        if (callInfo.Count < 2)
+        {
+            return scriptContext->GetLibrary()->GetUndefined();
+        }
+
+        int64 length;
+        Var iterable = args.Values[1];
+
+        TypedArrayBase *typedArrayBase = nullptr;
+        Assert(!JavascriptArray::Is(iterable));
+
+        if (TypedArrayBase::Is(iterable))
+        {
+            typedArrayBase = TypedArrayBase::FromVar(iterable);
+            if (typedArrayBase->IsDetachedBuffer())
+            {
+                JavascriptError::ThrowTypeError(scriptContext, JSERR_DetachedTypedArray);
+            }
+
+            length = typedArrayBase->GetLength();
+        }
+        else
+        {
+            length = JavascriptConversion::ToLength(JavascriptOperators::OP_GetLength(iterable, scriptContext), scriptContext);
+        }
+
+        return JavascriptNumber::ToVar(length, scriptContext);
+    }
+
+    /*
+    * First parameter is the object onto which prototype should be set; second is the value
+    */
+    Var JsBuiltInEngineInterfaceExtensionObject::EntryJsBuiltIn_Internal_SetPrototype(RecyclableObject *function, CallInfo callInfo, ...)
```

Comment:
Is there any way we can share native helpers like this between JsBuiltIns and Intl.js? I know there is a similar function in the Intl extension object.

Reply:
I can try moving it to  EngineInterfaceObject::InitializeCommonNativeInterfaces.

Reply:
I am going to address this as a separate PR. Also - https://github.com/Microsoft/ChakraCore/pull/3962 is incomplete and requires some special handling for EnginerInterfaceObject's deferred type handler.